python/tests/test_types.py

```diff
@@ -101,9 +101,7 @@ def test_struct_type():
     assert len(type_var.fields) == 3
     assert str(type_var) == str(eval(repr(type_var)))
     assert type_var == eval(repr(type_var))
-    assert type_var != StructType(
-        [NestedField(True, 1, "optional_field", IntegerType())]
-    )
+    assert type_var != StructType([NestedField(True, 1, "optional_field", IntegerType())])
```

Comment:
Can you add tests for `is_primitive`? I don't think those would be passing because many of the types aren't inheriting from `PrimitiveType`.

Reply:
added a test which tests all the types